### ISSUE TITLE
ShutdownMonitor should not be a singleton

### ISSUE THREAD

**Issue by sbordet** [eclipse/jetty.project#1820]
See discussion at #434.

`ShutdownMonitor` is a static singleton. We have a clearly defined lifecycle, so I think `ShutdownMonitor` can just become a normal non-static class that is added as a bean to `Server`.

**Comment by sbordet** [comment]
This issue should be fixed.

**Comment by sbordet** [comment]
@janbartel could you look at this?

**Comment by gregw** [comment]
@sbordet butting in here with a number of thoughts & questions.... 

So is the idea that if we have N servers running in a JVM, then there are N ports open listening for STOP commands?   
This is a bit strange as each one of them can issue a System.exit and shutdown the whole JVM... potentially not cleanly stopping the other servers.

If we change the ShutdownMonitor to just  stop one server it is associated and hope that causes the JVM to exit if there is only 1 server running, then we become vulnerable to one single non daemon thread stopping the thread from exiting and our external scripts are going to have to wait and then try a kill etc. (or do they do this anyway)?

So as much as I dislike statics, I think this is actually a use-case that makes sense for them.

Currently I'm thinking that `ShutdownMonitor` is doing an OK job of providing management for an entire JVM, which I think is useful.     Perhaps there is an alternate non static mechanism that can be applied to individual servers that just stop them and don't exit the JVM, but that feels like a different mechanism.... also do we actually have a use-case for that?  When does an external management script want to issue a command to stop just one server and not the JVM?

**Comment by joakime** [comment]
`ShutdownMonitor` is also tied to the hip of `ShutdownThread` which is directly related to `Runtime.getRuntime().addShutdownHook()`.

This Issue would result in changes across everything that uses `ShutdownThread` to know which instance specific `ShutdownMonitor` it belongs to as well.

**Comment by euvitudo** [comment]
One use case that may not have been considered is when a project has two war modules with different configurations that is built with mvnd.  When ShutdownMonitor is stopped when one of the modules finishes its integration tests, the other module fails the build since its integration tests cannot connect (as that particular instance of jetty is already shut down).

This project shows exhibits the issue I've encountered: https://github.com/euvitudo/mvnd-jetty-plugin-issue-2

This project has a patch that updates ShutdownMonitor and ShutdownThread to no longer be singletons: https://github.com/euvitudo/jetty.project (default branch is `jetty-10.0.x-mvnd-issue`).

Granted I am not as familiar with the full jetty.project codebase, hence there may be implications of which I am not aware.

Note: As an alternative, I've considered using the FORK deployment mode, but I've struggled with getting that to work for some reason (obviously another issue I haven't chased down).

I'd be happy to submit a pull request if desired.

**Comment by gregw** [comment]
@euvitudo That sound more like a use-case for something somewhere to do some reference counting and to only really shutdown after all the shutdown commands are issued.  
Or better yet, if the test module is running in the one JVM, then it should not be able to directly stop the server that it used, rather than issuing a shutdown JVM command to the monitor.

**Comment by euvitudo** [comment]
@gregw Thanks for the follow-up.

Thing is, the proof of concept project I referenced starts two servers with jetty-maven-plugin--one for each maven module. I would expect each module's server to be sufficiently isolated so as to not cause issues with any other modules.  That's what sort of surprised me when I migrated to this plugin using mvnd--it's not behavior that I expected and it certainly was not documented.

The reference counting seems like a bandaid to me, honestly.

Agree with your "better yet" comment--which again leads me to my opinion that ShutdownMonitor and ShutdownThread should not be singletons.

Thanks!

**Comment by gregw** [comment]
@euvitudo But ShutdownMonitor is precisely for shutting down the entire JVM.   That is it's job.   To stop a single server, then just server.stop should be called.   If this needs to be remote, then a JMX call can be done to stop a single server.

So again, I don't think the problem is with ShutdownMonitor being a singleton.  The problem is that it is being used when it should not be.    I'll defer to our maven plugin experts to comment on why this is so? @janbartel @olamy ?

**Comment by euvitudo** [comment]
@gregw Thanks for your explanation--it led me to consider the following.

The fundamental issue that led me down this path is that when using `mvnd` to build my (non-example) project that does have two integration-test modules that each use `jetty-maven-plugin` is the following message:

> `Error starting jetty: ShutdownMonitor already started`

I traced this down to the the second startup of a jetty server instance that had a different port/key configured.

I wonder if ShutdownMonitor could be split up such that you have a class that does the reference counting to determine whether the JVM should be shut down, then have ShutdownMonitors listen to the ports for the shutdown request, notifying the reference counter that that server has exited.  Just thinking aloud.

Thanks!

**Comment by janbartel** [comment]
@euvitudo I can't replicate your problem. For me, your project builds perfectly. Jetty starts and stops as expected. You are using jetty in an embedded mode, in that case "stop" does not stop the whole jvm, just the `Server` instance. The build proceeds sequentially, so a war is built, it's integration tests then executed and then the other war is built. So there is always only a single `Server` and `ShutdownMonitor` in existence at any one time.  Maybe your version of maven is different - mine is 3.8.4? Maybe somehow you are executing in parallel without telling the `failsafe` plugin to use forks? 

Here's the trace of my build:

> [33] mvn clean install
> [INFO] Scanning for projects...
> [INFO] ------------------------------------------------------------------------
> [INFO] Reactor Build Order:
> [INFO] 
> [INFO] mvnd-jetty-plugin-issue-2                                          [pom]
> [INFO] example-war-1                                                      [war]
> [INFO] example-war-2                                                      [war]
> [INFO] 
> [INFO] ---------------< org.example:mvnd-jetty-plugin-issue-2 >----------------
> [INFO] Building mvnd-jetty-plugin-issue-2 1.0-SNAPSHOT                    [1/3]
> [INFO] --------------------------------[ pom ]---------------------------------
> [INFO] 
> [INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ mvnd-jetty-plugin-issue-2 ---
> [INFO] 
> [INFO] --- gmavenplus-plugin:1.13.1:addSources (default) @ mvnd-jetty-plugin-issue-2 ---
> [INFO] 
> [INFO] --- gmavenplus-plugin:1.13.1:addTestSources (default) @ mvnd-jetty-plugin-issue-2 ---
> [INFO] 
> [INFO] --- gmavenplus-plugin:1.13.1:generateStubs (default) @ mvnd-jetty-plugin-issue-2 ---
> [INFO] No sources specified for stub generation. Skipping.
> [INFO] Generated 0 stubs.
> [INFO] 
> [INFO] --- gmavenplus-plugin:1.13.1:compile (default) @ mvnd-jetty-plugin-issue-2 ---
> [INFO] No sources specified for compilation. Skipping.
> [INFO] 
> [INFO] --- gmavenplus-plugin:1.13.1:removeStubs (default) @ mvnd-jetty-plugin-issue-2 ---
> [INFO] 
> [INFO] --- gmavenplus-plugin:1.13.1:generateTestStubs (default) @ mvnd-jetty-plugin-issue-2 ---
> [INFO] No sources specified for stub generation. Skipping.
> [INFO] Generated 0 stubs.
> [INFO] 
> [INFO] --- gmavenplus-plugin:1.13.1:compileTests (default) @ mvnd-jetty-plugin-issue-2 ---
> [INFO] No sources specified for compilation. Skipping.
> [INFO] 
> [INFO] --- gmavenplus-plugin:1.13.1:removeTestStubs (default) @ mvnd-jetty-plugin-issue-2 ---
> [INFO] 
> [INFO] --- maven-surefire-plugin:3.0.0-M7:test (default-test) @ mvnd-jetty-plugin-issue-2 ---
> [INFO] No tests to run.
> [INFO] 
> [INFO] --- maven-failsafe-plugin:3.0.0-M7:integration-test (functional-test) @ mvnd-jetty-plugin-issue-2 ---
> [INFO] No tests to run.
> [INFO] 
> [INFO] --- maven-failsafe-plugin:3.0.0-M7:verify (functional-test-verify) @ mvnd-jetty-plugin-issue-2 ---
> [INFO] 
> [INFO] --- maven-install-plugin:2.4:install (default-install) @ mvnd-jetty-plugin-issue-2 ---
> [INFO] Installing /home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/pom.xml to /home/janb/.m2/repository/org/example/mvnd-jetty-plugin-issue-2/1.0-SNAPSHOT/mvnd-jetty-plugin-issue-2-1.0-SNAPSHOT.pom
> [INFO] 
> [INFO] ---------------------< org.example:example-war-1 >----------------------
> [INFO] Building example-war-1 1.0-SNAPSHOT                                [2/3]
> [INFO] --------------------------------[ war ]---------------------------------
> [INFO] 
> [INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ example-war-1 ---
> [INFO] Deleting /home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-1/target
> [INFO] 
> [INFO] --- gmavenplus-plugin:1.13.1:addSources (default) @ example-war-1 ---
> [INFO] 
> [INFO] --- gmavenplus-plugin:1.13.1:addTestSources (default) @ example-war-1 ---
> [INFO] 
> [INFO] --- gmavenplus-plugin:1.13.1:generateStubs (default) @ example-war-1 ---
> [INFO] No sources specified for stub generation. Skipping.
> [INFO] Generated 0 stubs.
> [INFO] 
> [INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ example-war-1 ---
> [INFO] Using 'UTF-8' encoding to copy filtered resources.
> [INFO] skip non existing resourceDirectory /home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-1/src/main/resources
> [INFO] 
> [INFO] --- maven-compiler-plugin:3.10.1:compile (default-compile) @ example-war-1 ---
> [INFO] Changes detected - recompiling the module!
> [INFO] Compiling 1 source file to /home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-1/target/classes
> [INFO] 
> [INFO] --- gmavenplus-plugin:1.13.1:compile (default) @ example-war-1 ---
> [INFO] No sources specified for compilation. Skipping.
> [INFO] 
> [INFO] --- gmavenplus-plugin:1.13.1:removeStubs (default) @ example-war-1 ---
> [INFO] 
> [INFO] --- gmavenplus-plugin:1.13.1:generateTestStubs (default) @ example-war-1 ---
> [INFO] Using isolated classloader, without GMavenPlus classpath.
> [INFO] Using Groovy 3.0.12 to perform generateTestStubs.
> [INFO] Generated 1 stub.
> [INFO] 
> [INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ example-war-1 ---
> [INFO] Using 'UTF-8' encoding to copy filtered resources.
> [INFO] skip non existing resourceDirectory /home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-1/src/test/resources
> [INFO] 
> [INFO] --- maven-compiler-plugin:3.10.1:testCompile (default-testCompile) @ example-war-1 ---
> [INFO] Changes detected - recompiling the module!
> [INFO] Compiling 1 source file to /home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-1/target/test-classes
> [INFO] 
> [INFO] --- gmavenplus-plugin:1.13.1:compileTests (default) @ example-war-1 ---
> [INFO] Using isolated classloader, without GMavenPlus classpath.
> [INFO] Using Groovy 3.0.12 to perform compileTests.
> [INFO] Parallel parsing disabled.
> [INFO] Compiled 1 file.
> [INFO] 
> [INFO] --- gmavenplus-plugin:1.13.1:removeTestStubs (default) @ example-war-1 ---
> [INFO] 
> [INFO] --- maven-surefire-plugin:3.0.0-M7:test (default-test) @ example-war-1 ---
> [INFO] 
> [INFO] --- maven-war-plugin:3.3.2:war (default-war) @ example-war-1 ---
> [INFO] Packaging webapp
> [INFO] Assembling webapp [example-war-1] in [/home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-1/target/example-war-1-1.0-SNAPSHOT]
> [INFO] Processing war project
> [INFO] Copying webapp resources [/home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-1/src/main/webapp]
> [INFO] Building war: /home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-1/target/example-war-1.war
> [INFO] 
> [INFO] >>> jetty-maven-plugin:10.0.12:start (start-jetty) > validate @ example-war-1 >>>
> [INFO] 
> [INFO] <<< jetty-maven-plugin:10.0.12:start (start-jetty) < validate @ example-war-1 <<<
> [INFO] 
> [INFO] 
> [INFO] --- jetty-maven-plugin:10.0.12:start (start-jetty) @ example-war-1 ---
> [INFO] Configuring Jetty for project: example-war-1
> [INFO] Classes = /home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-1/target/classes
> [INFO] Context path = /mvnd-test
> [INFO] Tmp directory = /home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-1/target/tmp
> [INFO] web.xml file = file:///home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-1/src/main/webapp/WEB-INF/web.xml
> [INFO] Webapp directory = /home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-1/src/main/webapp
> [INFO] Web defaults = org/eclipse/jetty/webapp/webdefault.xml
> [INFO] Web overrides =  none
> [INFO] jetty-10.0.12; built: 2022-09-14T01:54:40.076Z; git: 408d0139887e27a57b54ed52e2d92a36731a7e88; jvm 17+35-2724
> [WARNING] javax.activation.ActivationDataFlavor scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/ActivationDataFlavor.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/ActivationDataFlavor.class
> [WARNING] javax.activation.CommandInfo$Beans$1 scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/CommandInfo$Beans$1.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/CommandInfo$Beans$1.class
> [WARNING] javax.activation.CommandInfo$Beans scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/CommandInfo$Beans.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/CommandInfo$Beans.class
> [WARNING] javax.activation.CommandInfo scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/CommandInfo.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/CommandInfo.class
> [WARNING] javax.activation.CommandMap scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/CommandMap.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/CommandMap.class
> [WARNING] javax.activation.CommandObject scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/CommandObject.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/CommandObject.class
> [WARNING] javax.activation.DataContentHandler scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/DataContentHandler.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/DataContentHandler.class
> [WARNING] javax.activation.DataContentHandlerFactory scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/DataContentHandlerFactory.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/DataContentHandlerFactory.class
> [WARNING] javax.activation.DataHandler$1 scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/DataHandler$1.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/DataHandler$1.class
> [WARNING] javax.activation.DataHandler scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/DataHandler.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/DataHandler.class
> [WARNING] javax.activation.DataHandlerDataSource scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/DataHandlerDataSource.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/DataHandlerDataSource.class
> [WARNING] javax.activation.DataSource scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/DataSource.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/DataSource.class
> [WARNING] javax.activation.DataSourceDataContentHandler scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/DataSourceDataContentHandler.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/DataSourceDataContentHandler.class
> [WARNING] javax.activation.FileDataSource scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/FileDataSource.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/FileDataSource.class
> [WARNING] javax.activation.FileTypeMap scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/FileTypeMap.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/FileTypeMap.class
> [WARNING] javax.activation.MailcapCommandMap$1 scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/MailcapCommandMap$1.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/MailcapCommandMap$1.class
> [WARNING] javax.activation.MailcapCommandMap scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/MailcapCommandMap.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/MailcapCommandMap.class
> [WARNING] javax.activation.MimeType scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/MimeType.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/MimeType.class
> [WARNING] javax.activation.MimeTypeParameterList scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/MimeTypeParameterList.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/MimeTypeParameterList.class
> [WARNING] javax.activation.MimeTypeParseException scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/MimeTypeParseException.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/MimeTypeParseException.class
> [WARNING] javax.activation.MimetypesFileTypeMap$1 scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/MimetypesFileTypeMap$1.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/MimetypesFileTypeMap$1.class
> [WARNING] javax.activation.MimetypesFileTypeMap scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/MimetypesFileTypeMap.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/MimetypesFileTypeMap.class
> [WARNING] javax.activation.ObjectDataContentHandler scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/ObjectDataContentHandler.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/ObjectDataContentHandler.class
> [WARNING] javax.activation.SecuritySupport$1 scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/SecuritySupport$1.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/SecuritySupport$1.class
> [WARNING] javax.activation.SecuritySupport$2 scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/SecuritySupport$2.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/SecuritySupport$2.class
> [WARNING] javax.activation.SecuritySupport$3 scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/SecuritySupport$3.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/SecuritySupport$3.class
> [WARNING] javax.activation.SecuritySupport$4 scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/SecuritySupport$4.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/SecuritySupport$4.class
> [WARNING] javax.activation.SecuritySupport$5 scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/SecuritySupport$5.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/SecuritySupport$5.class
> [WARNING] javax.activation.SecuritySupport scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/SecuritySupport.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/SecuritySupport.class
> [WARNING] javax.activation.URLDataSource scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/URLDataSource.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/URLDataSource.class
> [WARNING] javax.activation.UnsupportedDataTypeException scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/UnsupportedDataTypeException.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/UnsupportedDataTypeException.class
> [INFO] No Spring WebApplicationInitializer types detected on classpath
> [INFO] Session workerName=node0
> [INFO] Initializing Spring root WebApplicationContext
> [INFO] Root WebApplicationContext: initialization started
> [INFO] Root WebApplicationContext initialized in 272 ms
> [INFO] Started o.e.j.m.p.MavenWebAppContext@3a42145{/mvnd-test,[file:///home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-1/src/main/webapp/],AVAILABLE}{file:///home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-1/src/main/webapp/}
> [INFO] Started ServerConnector@427c2c96{HTTP/1.1, (http/1.1)}{127.0.0.1:18080}
> [INFO] Started Server@6055e899{STARTING}[10.0.12,sto=0] @6766ms
> [INFO] 
> [INFO] --- maven-failsafe-plugin:3.0.0-M7:integration-test (functional-test) @ example-war-1 ---
> [INFO] Using auto detected provider org.apache.maven.surefire.junitplatform.JUnitPlatformProvider
> [INFO] 
> [INFO] -------------------------------------------------------
> [INFO]  T E S T S
> [INFO] -------------------------------------------------------
> [INFO] Running com.example.TestFunctionalSpec
> [INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.334 s - in com.example.TestFunctionalSpec
> [INFO] 
> [INFO] Results:
> [INFO] 
> [INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
> [INFO] 
> [INFO] 
> [INFO] --- jetty-maven-plugin:10.0.12:stop (stop-jetty) @ example-war-1 ---
> [INFO] Stopping jetty
> [INFO] Stopped Server@6055e899{STOPPING}[10.0.12,sto=0]
> [INFO] Stopped ServerConnector@427c2c96{HTTP/1.1, (http/1.1)}{127.0.0.1:18080}
> [INFO] Closing Spring root WebApplicationContext
> [INFO] Stopped o.e.j.m.p.MavenWebAppContext@3a42145{/mvnd-test,[file:///home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-1/src/main/webapp/],STOPPED}{file:///home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-1/src/main/webapp/}
> [INFO] 
> [INFO] --- maven-failsafe-plugin:3.0.0-M7:verify (functional-test-verify) @ example-war-1 ---
> [INFO] 
> [INFO] --- maven-install-plugin:2.4:install (default-install) @ example-war-1 ---
> [INFO] Installing /home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-1/target/example-war-1.war to /home/janb/.m2/repository/org/example/example-war-1/1.0-SNAPSHOT/example-war-1-1.0-SNAPSHOT.war
> [INFO] Installing /home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-1/pom.xml to /home/janb/.m2/repository/org/example/example-war-1/1.0-SNAPSHOT/example-war-1-1.0-SNAPSHOT.pom
> [INFO] 
> [INFO] ---------------------< org.example:example-war-2 >----------------------
> [INFO] Building example-war-2 1.0-SNAPSHOT                                [3/3]
> [INFO] --------------------------------[ war ]---------------------------------
> [INFO] 
> [INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ example-war-2 ---
> [INFO] Deleting /home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-2/target
> [INFO] 
> [INFO] --- gmavenplus-plugin:1.13.1:addSources (default) @ example-war-2 ---
> [INFO] 
> [INFO] --- gmavenplus-plugin:1.13.1:addTestSources (default) @ example-war-2 ---
> [INFO] 
> [INFO] --- gmavenplus-plugin:1.13.1:generateStubs (default) @ example-war-2 ---
> [INFO] No sources specified for stub generation. Skipping.
> [INFO] Generated 0 stubs.
> [INFO] 
> [INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ example-war-2 ---
> [INFO] Using 'UTF-8' encoding to copy filtered resources.
> [INFO] skip non existing resourceDirectory /home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-2/src/main/resources
> [INFO] 
> [INFO] --- maven-compiler-plugin:3.10.1:compile (default-compile) @ example-war-2 ---
> [INFO] Changes detected - recompiling the module!
> [INFO] Compiling 1 source file to /home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-2/target/classes
> [INFO] 
> [INFO] --- gmavenplus-plugin:1.13.1:compile (default) @ example-war-2 ---
> [INFO] No sources specified for compilation. Skipping.
> [INFO] 
> [INFO] --- gmavenplus-plugin:1.13.1:removeStubs (default) @ example-war-2 ---
> [INFO] 
> [INFO] --- gmavenplus-plugin:1.13.1:generateTestStubs (default) @ example-war-2 ---
> [INFO] Using isolated classloader, without GMavenPlus classpath.
> [INFO] Using Groovy 3.0.12 to perform generateTestStubs.
> [INFO] Generated 1 stub.
> [INFO] 
> [INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ example-war-2 ---
> [INFO] Using 'UTF-8' encoding to copy filtered resources.
> [INFO] skip non existing resourceDirectory /home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-2/src/test/resources
> [INFO] 
> [INFO] --- maven-compiler-plugin:3.10.1:testCompile (default-testCompile) @ example-war-2 ---
> [INFO] Changes detected - recompiling the module!
> [INFO] Compiling 1 source file to /home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-2/target/test-classes
> [INFO] 
> [INFO] --- gmavenplus-plugin:1.13.1:compileTests (default) @ example-war-2 ---
> [INFO] Using isolated classloader, without GMavenPlus classpath.
> [INFO] Using Groovy 3.0.12 to perform compileTests.
> [INFO] Parallel parsing disabled.
> [INFO] Compiled 1 file.
> [INFO] 
> [INFO] --- gmavenplus-plugin:1.13.1:removeTestStubs (default) @ example-war-2 ---
> [INFO] 
> [INFO] --- maven-surefire-plugin:3.0.0-M7:test (default-test) @ example-war-2 ---
> [INFO] 
> [INFO] --- maven-war-plugin:3.3.2:war (default-war) @ example-war-2 ---
> [INFO] Packaging webapp
> [INFO] Assembling webapp [example-war-2] in [/home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-2/target/example-war-2-1.0-SNAPSHOT]
> [INFO] Processing war project
> [INFO] Copying webapp resources [/home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-2/src/main/webapp]
> [INFO] Building war: /home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-2/target/example-war-2.war
> [INFO] 
> [INFO] >>> jetty-maven-plugin:10.0.12:start (start-jetty) > validate @ example-war-2 >>>
> [INFO] 
> [INFO] <<< jetty-maven-plugin:10.0.12:start (start-jetty) < validate @ example-war-2 <<<
> [INFO] 
> [INFO] 
> [INFO] --- jetty-maven-plugin:10.0.12:start (start-jetty) @ example-war-2 ---
> [INFO] Configuring Jetty for project: example-war-2
> [INFO] Classes = /home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-2/target/classes
> [INFO] Context path = /mvnd-test
> [INFO] Tmp directory = /home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-2/target/tmp
> [INFO] web.xml file = file:///home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-2/src/main/webapp/WEB-INF/web.xml
> [INFO] Webapp directory = /home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-2/src/main/webapp
> [INFO] Web defaults = org/eclipse/jetty/webapp/webdefault.xml
> [INFO] Web overrides =  none
> [INFO] jetty-10.0.12; built: 2022-09-14T01:54:40.076Z; git: 408d0139887e27a57b54ed52e2d92a36731a7e88; jvm 17+35-2724
> [WARNING] javax.activation.ActivationDataFlavor scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/ActivationDataFlavor.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/ActivationDataFlavor.class
> [WARNING] javax.activation.CommandInfo$Beans$1 scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/CommandInfo$Beans$1.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/CommandInfo$Beans$1.class
> [WARNING] javax.activation.CommandInfo$Beans scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/CommandInfo$Beans.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/CommandInfo$Beans.class
> [WARNING] javax.activation.CommandInfo scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/CommandInfo.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/CommandInfo.class
> [WARNING] javax.activation.CommandMap scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/CommandMap.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/CommandMap.class
> [WARNING] javax.activation.CommandObject scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/CommandObject.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/CommandObject.class
> [WARNING] javax.activation.DataContentHandler scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/DataContentHandler.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/DataContentHandler.class
> [WARNING] javax.activation.DataContentHandlerFactory scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/DataContentHandlerFactory.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/DataContentHandlerFactory.class
> [WARNING] javax.activation.DataHandler$1 scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/DataHandler$1.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/DataHandler$1.class
> [WARNING] javax.activation.DataHandler scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/DataHandler.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/DataHandler.class
> [WARNING] javax.activation.DataHandlerDataSource scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/DataHandlerDataSource.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/DataHandlerDataSource.class
> [WARNING] javax.activation.DataSource scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/DataSource.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/DataSource.class
> [WARNING] javax.activation.DataSourceDataContentHandler scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/DataSourceDataContentHandler.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/DataSourceDataContentHandler.class
> [WARNING] javax.activation.FileDataSource scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/FileDataSource.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/FileDataSource.class
> [WARNING] javax.activation.FileTypeMap scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/FileTypeMap.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/FileTypeMap.class
> [WARNING] javax.activation.MailcapCommandMap$1 scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/MailcapCommandMap$1.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/MailcapCommandMap$1.class
> [WARNING] javax.activation.MailcapCommandMap scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/MailcapCommandMap.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/MailcapCommandMap.class
> [WARNING] javax.activation.MimeType scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/MimeType.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/MimeType.class
> [WARNING] javax.activation.MimeTypeParameterList scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/MimeTypeParameterList.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/MimeTypeParameterList.class
> [WARNING] javax.activation.MimeTypeParseException scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/MimeTypeParseException.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/MimeTypeParseException.class
> [WARNING] javax.activation.MimetypesFileTypeMap$1 scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/MimetypesFileTypeMap$1.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/MimetypesFileTypeMap$1.class
> [WARNING] javax.activation.MimetypesFileTypeMap scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/MimetypesFileTypeMap.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/MimetypesFileTypeMap.class
> [WARNING] javax.activation.ObjectDataContentHandler scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/ObjectDataContentHandler.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/ObjectDataContentHandler.class
> [WARNING] javax.activation.SecuritySupport$1 scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/SecuritySupport$1.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/SecuritySupport$1.class
> [WARNING] javax.activation.SecuritySupport$2 scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/SecuritySupport$2.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/SecuritySupport$2.class
> [WARNING] javax.activation.SecuritySupport$3 scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/SecuritySupport$3.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/SecuritySupport$3.class
> [WARNING] javax.activation.SecuritySupport$4 scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/SecuritySupport$4.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/SecuritySupport$4.class
> [WARNING] javax.activation.SecuritySupport$5 scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/SecuritySupport$5.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/SecuritySupport$5.class
> [WARNING] javax.activation.SecuritySupport scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/SecuritySupport.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/SecuritySupport.class
> [WARNING] javax.activation.URLDataSource scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/URLDataSource.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/URLDataSource.class
> [WARNING] javax.activation.UnsupportedDataTypeException scanned from multiple locations: jar:file:///home/janb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar!/javax/activation/UnsupportedDataTypeException.class, jar:file:///home/janb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar!/javax/activation/UnsupportedDataTypeException.class
> [INFO] No Spring WebApplicationInitializer types detected on classpath
> [INFO] Session workerName=node0
> [INFO] Initializing Spring root WebApplicationContext
> [INFO] Root WebApplicationContext: initialization started
> [INFO] Root WebApplicationContext initialized in 169 ms
> [INFO] Started o.e.j.m.p.MavenWebAppContext@4398d{/mvnd-test,[file:///home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-2/src/main/webapp/],AVAILABLE}{file:///home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-2/src/main/webapp/}
> [INFO] Started ServerConnector@239fdf8f{HTTP/1.1, (http/1.1)}{127.0.0.1:28080}
> [INFO] Started Server@18a8a40e{STARTING}[10.0.12,sto=0] @10658ms
> [INFO] 
> [INFO] --- maven-failsafe-plugin:3.0.0-M7:integration-test (functional-test) @ example-war-2 ---
> [INFO] Using auto detected provider org.apache.maven.surefire.junitplatform.JUnitPlatformProvider
> [INFO] 
> [INFO] -------------------------------------------------------
> [INFO]  T E S T S
> [INFO] -------------------------------------------------------
> [INFO] Running com.example.TestFunctionalSpec
> [INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.268 s - in com.example.TestFunctionalSpec
> [INFO] 
> [INFO] Results:
> [INFO] 
> [INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
> [INFO] 
> [INFO] 
> [INFO] --- jetty-maven-plugin:10.0.12:stop (stop-jetty) @ example-war-2 ---
> [INFO] Stopping jetty
> [INFO] Stopped Server@18a8a40e{STOPPING}[10.0.12,sto=0]
> [INFO] Stopped ServerConnector@239fdf8f{HTTP/1.1, (http/1.1)}{127.0.0.1:28080}
> [INFO] Closing Spring root WebApplicationContext
> [INFO] Stopped o.e.j.m.p.MavenWebAppContext@4398d{/mvnd-test,[file:///home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-2/src/main/webapp/],STOPPED}{file:///home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-2/src/main/webapp/}
> [INFO] 
> [INFO] --- maven-failsafe-plugin:3.0.0-M7:verify (functional-test-verify) @ example-war-2 ---
> [INFO] 
> [INFO] --- maven-install-plugin:2.4:install (default-install) @ example-war-2 ---
> [INFO] Installing /home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-2/target/example-war-2.war to /home/janb/.m2/repository/org/example/example-war-2/1.0-SNAPSHOT/example-war-2-1.0-SNAPSHOT.war
> [INFO] Installing /home/janb/src/tmp/issue-1820/mvnd-jetty-plugin-issue-2-main/example-war-2/pom.xml to /home/janb/.m2/repository/org/example/example-war-2/1.0-SNAPSHOT/example-war-2-1.0-SNAPSHOT.pom
> [INFO] ------------------------------------------------------------------------
> [INFO] Reactor Summary for mvnd-jetty-plugin-issue-2 1.0-SNAPSHOT:
> [INFO] 
> [INFO] mvnd-jetty-plugin-issue-2 .......................... SUCCESS [  1.073 s]
> [INFO] example-war-1 ...................................... SUCCESS [  6.510 s]
> [INFO] example-war-2 ...................................... SUCCESS [  3.417 s]
> [INFO] ------------------------------------------------------------------------
> [INFO] BUILD SUCCESS
> [INFO] ------------------------------------------------------------------------
> [INFO] Total time:  11.133 s
> [INFO] Finished at: 2022-10-04T09:47:39+11:00
> [INFO] ------------------------------------------------------------------------
>

**Comment by olamy** [comment]
@janbartel to reproduce you need to use `mvnd` not `mvn` ;) 
https://github.com/apache/maven-mvnd

this is reusing jvm to avoid startup time

**Comment by janbartel** [comment]
@olamy well there's your problem then ;)

**Comment by euvitudo** [comment]
@janbartel Yes, I realize that's the problem.  It wasn't clear from the documentation that I *shouldn't* necessarily be running two instances in the same JVM (i.e., not using deployMode=FORK).  If it's there, I simply missed it and would be grateful to be pointed to the location.

If the original suggestion by @sbordet to migrate ShutdownMonitor from a singleton were implemented (assuming the various caveats and/or possible implementations discussed here are even remotely helpful), then it would be possible to simultaneously run two (or possibly more) integration-test servers in the JVM using `jetty-maven-plugin`, with each usage starting and stopping their respective servers at the obvious times.

Notwithstanding the original issue I had with mvnd (namely the singleton issue discussed here), I have things working with `deployMode=FORK`, though I had to argue with the configuration a bit, as the documentation was not consistent with what ended up working.

Thanks!

**Comment by olamy** [comment]
@euvitudo if using `FORK` is fine for you. Can we consider this as fixed?

**Comment by janbartel** [comment]
@euvitudo any info on where the documentation needs improving would be gratefully accepted.

**Comment by euvitudo** [comment]
@olamy Considering I'm not the one who originally created this issue, I'm not sure I'm the one who can say whether to close it.  I did end up highjacking it a bit, but I thought it was still relevant the original issue.  Maybe ask @sbordet?

Regarding the `jetty-maven-plugin` and the use of `FORK`, I'm fine running with that for now.

@janbartel I'll compose some notes and create a new issue.  Thanks!

**Comment by olamy** [comment]
@euvitudo when this issue "ShutdownMonitor should not be a singleton" will be fixed. `mvnd` will not be a problem anymore when using non `FORK` :)
